### PR TITLE
Feature/wallet modal DAO-436 

### PR DESCRIPTION
**Pull request by sepehr2github**
- Wallet Modal added
- Wallet Card added
- Fix Bottom sheet fixed height issue
- Dynamic navbar wallet button added
- Identicon Avatar added

### CODE REVIEW

File: packages/ui-components/stories/WalletCard.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {Meta, Story} from '@storybook/react';
+import {WalletCard, WalletCardProps} from '../src';
+
+export default {
+  title: 'Components/WalletCardProps',
```
Comment (WissenIstNacht) [inline packages/ui-components/stories/WalletCard.stories.tsx]:
```suggestion
  title: 'Components/WalletCard',
```

File: packages/ui-components/src/components/walletCard/walletCard.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  /**
+  * wallet ENS name or wallet eth address
+  */
+  name: string;
+  /**
+  * Wallet eth address
+  */
+  address?: string | null;
+  /**
+  * Allows the Wallet Card component grow vertically
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/walletCard/walletCard.tsx]: horizontally? :p

Edit: I guess that on me haha

File: packages/ui-components/src/components/walletCard/walletCard.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            mode={'ghost'} 
+            label="copy" 
+            side="right" 
+            onClick={copyToClipboard}
+          />
+      </Card>
+  );
+};
+
+type ContainerProps = Pick<WalletCardProps, 'wide'>;
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/walletCard/walletCard.tsx:53]: Nice. I think I forgot to do that on my last PR about `DaoCard`.

File: packages/ui-components/src/components/walletCard/walletCard.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  const copyToClipboard = () => {
+    navigator.clipboard.writeText(address || name);
+  }
+
+  return (
+      <Card {...{wide}} data-testid="walletCard">
+          <Content>
+            <Avatar size={'default'} src={src}/>
+            <TextContainer>
+              <Title>{BeautifyLabel(name)}</Title>
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/walletCard/walletCard.tsx]: Care, passing `sdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdfasdf.eth` will not shorten the addres. 

No need to change anything right now. AFAIK the designs don't really specify anything about how to shorten ens names yet.

File: packages/ui-components/src/components/walletCard/walletCard.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import styled from 'styled-components';
+
+import {Avatar,AvatarProps} from '../avatar';
+import {IconButton} from '../button/iconButton';
+import {IconCopy} from '../icons'
+import {BeautifyLabel} from '../../utils/addresses';
+
+export type WalletCardProps = {
+  /**
+  * wallet ENS name or wallet eth address
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/walletCard/walletCard.tsx:11]: I understand why you did this, but I don't think it's the right approach. In the designs, the wallet address must always be present. I think this should be reflected in the interface. I would make `address` non-optional and `name` optional. Then
- if: `name` is not specified: make the `address` a `Title` 
- Else: leave it as you have it now

File: packages/ui-components/src/components/button/iconButton.tsx [hunk 6]
```diff
@@ -45,14 +49,10 @@ type FlexDivProps = {
 export const FlexDiv = styled.div.attrs(({side}: FlexDivProps) => {
   let className = 'flex items-center space-x-2';
   if (side === 'right')
-    className = 'flex items-center flex-row-reverse space-x-reverse space-x-2';
+    className = 'flex items-center flex-row-reverse space-x-reverse space-x-1.5';
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/button/iconButton.tsx:52]: Here you changed to `2.5`, but on line 50 you left to `2`. Is that intentional?

File: packages/web-app/src/containers/walletMenu/index.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      isOpen={isOpen}
+      onOpen={open}
+      onClose={close}
+      data-testid="walletCard"
+    >
+      <Container>
+        <WalletCard
+          wide
+          src={'https://place-hold.it/150x150'}
+          title="ens-name.eth"
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/walletMenu/index.tsx]:
```suggestion
          name="ens-name.eth"
```

File: packages/web-app/src/containers/walletMenu/index.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+          subtitle={account}
+        />
+        <ActionContainer>
+          <ActionListItem
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/walletMenu/index.tsx:28]: Still necessary? (seeing as we're only doing Metamask right now?)

File: packages/web-app/src/containers/navbar/index.tsx [hunk 9]
```diff
@@ -34,10 +43,11 @@ const Navbar: React.FC = () => {
             </StyledNavLink>
           </LinksContainer>
         </Container>
-        <AccountButton>
-          <div>punk420.eth</div>
-          <TempAvatar />
-        </AccountButton>
+        <WalletButton
+          onClick={onToggle}
+          label={isConnected() ? account : 'Connect account'}
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/navbar/index.tsx]: We can do this later, but technically, the `Connect account` label should also be part of the i10n translation.

**Comment by WissenIstNacht** [comment]
Nice one, this really takes us a step closer to a basic app! 🥳

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/walletCard/walletCard.tsx (hunk 2 above)
Comment (sepehr2github) [reply packages/ui-components/src/components/walletCard/walletCard.tsx]: Yes. I used your comment hahaha. I want to keep too much consistency LOL

File: packages/ui-components/__tests__/walletCard.test.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+
+import {Default as DaoCard} from '../stories/walletCard.stories';
+
+describe('DaoSelector', () => {
```
Comment (Fabricevladimir) [inline packages/ui-components/__tests__/walletCard.test.tsx]: Copy paste error?

File: packages/ui-components/src/components/button/walletButton.tsx [hunk 11]
```diff
@@ -1,60 +1,60 @@
...
+  );
+};
+
+type StyledButtonProp = Pick<WalletButtonProps, 'isSelected'>;
+type StyledLabelProp = Pick<WalletButtonProps, 'isLoading'>;
+
+const StyledButton = styled(SizedButton).attrs(
+  ({isSelected}: StyledButtonProp) => ({
+    className: `flex tablet:space-x-1.5 py-1.5 
+  ${isSelected ? 'text-primary-500 bg-primary-50' : 'text-ui-600 bg-ui-0'}`,
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/button/walletButton.tsx]: Can we get an `active` state for this as well for when the button is clicked?

File: packages/web-app/src/containers/navbar/index.tsx [hunk 12]
```diff
@@ -2,12 +2,21 @@ import React from 'react';
 import styled from 'styled-components';
 import {NavLink} from 'react-router-dom';
 import {useTranslation} from 'react-i18next';
+import {WalletButton} from '@aragon/ui-components';
 
 import TestNetworkIndicator from 'components/testNetworkIndicator';
 import {Dashboard, Community, Finance, Governance} from 'utils/paths';
+import {useMenuContext} from '../../context/menu';
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/navbar/index.tsx]: Can you switch this from a "relative" import?

File: packages/web-app/src/pages/home.tsx [hunk 13]
```diff
@@ -29,6 +30,7 @@ const Home: React.FC = () => {
       </div>
       <h1>Wallet</h1>
       <Wallet />
+      <WalletMenu />
       <Button onClick={onOpen} label="Toggle" />
       <BottomSheet onOpen={onOpen} isOpen={isOpen} onClose={onClose} />
```
Comment (Fabricevladimir) [inline packages/web-app/src/pages/home.tsx:35]: ~~I don't foresee a case where we would need multiple instances of the `BottomSheet` can this be moved to a central location like in `app` perhaps?~~

I might be thinking about this incorrectly. Disregard.

**Review by Fabricevladimir — CHANGES_REQUESTED** [review]
Nice work!

File: packages/ui-components/src/components/button/iconButton.tsx (hunk 6 above)
Comment (sepehr2github) [reply packages/ui-components/src/components/button/iconButton.tsx:52]: Yes, It's intentional. there is a difference in space between the right & left icon sides.

File: packages/ui-components/src/components/walletCard/walletCard.tsx (hunk 2 above)
Comment (WissenIstNacht) [reply packages/ui-components/src/components/walletCard/walletCard.tsx]: 👌🏼 😄 

Could you change mine to "horizontally", too? :)

File: packages/ui-components/src/components/button/walletButton.tsx [hunk 14]
```diff
@@ -1,60 +1,60 @@
...
+  );
+};
+
+type StyledButtonProp = Pick<WalletButtonProps, 'active'>;
+type StyledLabelProp = Pick<WalletButtonProps, 'isLoading'>;
+
+const StyledButton = styled(SizedButton).attrs(
+  ({active}: StyledButtonProp) => ({
+    className: `flex tablet:space-x-1.5
+      ${active ? 'text-primary-500 bg-primary-50' : 'text-ui-600 bg-ui-0'}`,
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/button/walletButton.tsx]: I might have been unclear a bit with this. What I meant was active in terms of state like `hover` and `disable`

**Review by Fabricevladimir — APPROVED** [review]

**Review by WissenIstNacht — APPROVED** [review]

**Comment by Fabricevladimir** [comment]
Closed for commit signing issues

**Pull request closed without merging**